api-registration/server/server.js

```diff
@@ -0,0 +1,45 @@
+import 'dotenv/config';
+import pg from 'pg';
+import argon2 from 'argon2';
+import express from 'express';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+app.use(express.json());
+app.post('/api/auth/sign-up', async (req, res, next) => {
+  try {
+    const { username, password } = req.body;
+    if (!username || !password) {
+      throw new ClientError(400, 'username and password are required fields');
+    }
+    const sql = `
+                insert into "users" ("username", "hashedPassword")
+                values ($1, $2)
+                returning *;
```

Comment:
Or you could return just the columns you are going to respond with. Then you wouldn't need to extract them at line 30.